go-controller/pkg/ovn/egressqos.go

```diff
@@ -816,14 +816,18 @@ func (oc *DefaultNetworkController) onEgressQoSPodUpdate(oldObj, newObj interfac
 	newPodLabels := labels.Set(newPod.Labels)
 	oldPodIPs, _ := util.GetPodIPsOfNetwork(oldPod, oc.NetInfo)
 	newPodIPs, _ := util.GetPodIPsOfNetwork(newPod, oc.NetInfo)
-	isOldPodLocal := oc.isPodScheduledinLocalZone(oldPod)
 	isNewPodLocal := oc.isPodScheduledinLocalZone(newPod)
 	oldPodCompleted := util.PodCompleted(oldPod)
 	newPodCompleted := util.PodCompleted(newPod)
+
+	// NOTE: We only expect remote pods to become local when they are scheduled; not vice versa
+	// Ignore the pod if it is remote to this zone.
+	if !isNewPodLocal {
```

Comment:
haha also smells like missing unit test coverage? can you add one?

Reply:
@tssurya added an unit test to cover this. PTAL.